### PR TITLE
feat: add share dialog

### PR DESCRIPTION
**Pull request by etouyang**
## Description

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/plugins/Pets/SNSAdaptor/PetSetDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+    const { t } = useI18N()
+    const classes = useStylesExtends(useStyles(), {})
+    const { showSnackbar } = useCustomSnackbar()
+    const [loading, setLoading] = useState(false)
+
+    const user = useUser()
+    const nfts = useNFTs(user, configNFTs)
+    const extraData = useNFTsExtra(configNFTs)
+    const [collection, setCollection] = useState<FilterContract>(initCollection)
+    const [isCollectionsError, setCollectionsError] = useState(false)
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Pets/SNSAdaptor/PetSetDialog.tsx]:
```suggestion
    const [isCollectionsError, setIsCollectionsError] = useState(false)
```

File: packages/mask/src/plugins/Pets/SNSAdaptor/PetSetDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+    const [metaData, setMetaData] = useState<PetMetaDB>(initMeta)
+    const [isImageError, setImageError] = useState(false)
+    const [isTipShow, setTipShow] = useState(false)
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Pets/SNSAdaptor/PetSetDialog.tsx]:
```suggestion
    const [isImageError, setIsImageError] = useState(false)
    const [isTipShow, setIsTipShow] = useState(false)
```

File: packages/mask/src/plugins/Pets/constants.ts [hunk 3]
```diff
@@ -34,6 +34,11 @@ export const Punk3D = {
     }),
 }
 
+export const Share_Twitter = 'https://twitter.com/realMaskNetwork/status/1486648872558424064?s=20'
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Pets/constants.ts]:
```suggestion
export const Share_Twitter = 'https://twitter.com/realMaskNetwork/status/1486648872558424064'
```

**Comment by septs** [comment]
plz sync develop branch

**Review by guanbinrui — DISMISSED** [review]

File: packages/mask/src/plugins/Pets/SNSAdaptor/PetSetDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+    const [holderChange, setHolderChange] = useState(true)
+    const [tokenInfoSelect, setTokenInfoSelect] = useState<OwnerERC721TokenInfo | null>(null)
+    const [inputTokenName, setInputTokenName] = useState('')
+
+    useEffect(() => {
+        setMetaData(initMeta)
+        setCollection(initCollection)
+        setTokenInfoSelect(null)
+        setInputTokenName('')
+    }, [])
```
Comment (Jack-Works) [inline packages/mask/src/plugins/Pets/SNSAdaptor/PetSetDialog.tsx]: Why did you need to do this? They're already set as the default value you have specified.

File: packages/mask/src/plugins/Pets/SNSAdaptor/PetSetDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+
+    let timer: NodeJS.Timeout | null = null
```
Comment (Jack-Works) [inline packages/mask/src/plugins/Pets/SNSAdaptor/PetSetDialog.tsx]: No. This isn't correct. You must store it in a `useRef` otherwise it won't work if the component gets rerendered.

File: packages/mask/src/plugins/Pets/SNSAdaptor/PetSetDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+    const closeDialogHandle = () => {
+        setIsTipShow(true)
+        onClose()
+        if (timer !== null) clearTimeout(timer)
```
Comment (Jack-Works) [inline packages/mask/src/plugins/Pets/SNSAdaptor/PetSetDialog.tsx]: Please try if `useTimeout` in `react-use` works for you.

File: packages/mask/src/plugins/Pets/SNSAdaptor/PetShareDialog.tsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    },
+}))
+
+interface PetSetDialogProps {
+    onClose: () => void
+}
+
+export function PetShareDialog({ onClose }: PetSetDialogProps) {
+    const { t } = useI18N()
+    const classes = useStylesExtends(useStyles(), {})
```
Comment (Jack-Works) [inline packages/mask/src/plugins/Pets/SNSAdaptor/PetShareDialog.tsx]:
```suggestion
    const { classes } = useStyles()
```

**Review by Jack-Works — CHANGES_REQUESTED** [review]

File: packages/mask/src/plugins/Pets/SNSAdaptor/PetSetDialog.tsx (hunk 4 above)
Comment (etouyang) [reply packages/mask/src/plugins/Pets/SNSAdaptor/PetSetDialog.tsx]: while close the dialog, state data did not expect

Comment (Jack-Works) [reply packages/mask/src/plugins/Pets/SNSAdaptor/PetSetDialog.tsx]: what do you mean?

Comment (etouyang) [reply packages/mask/src/plugins/Pets/SNSAdaptor/PetSetDialog.tsx]: yes, this code is redundant

**Review by guanbinrui — APPROVED** [review]
